actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -364,6 +375,16 @@ def multiple_sources_tag_builder(type, sources)
           end
         end
 
+        def resolve_image_source(source, skip_pipeline)
+          if source.is_a?(Symbol) || source.is_a?(String)
```

Comment:
We need to change `ActiveStorage::Variant` to respond to `to_model` instead. If we are expecting to use that class with URLs it needs to follow the `ActiveModel::Model` API.